mws/resource_mws_credentials.go

```diff
@@ -59,71 +29,81 @@ func (a CredentialsAPI) List(mwsAcctID string) ([]Credentials, error) {
 	return mwsCredsList, err
 }
 
+type CredentialInfo struct {
+	// The human-readable name of the credential configuration object.
+	AccountId string `json:"account_id,omitempty"`
+	// The human-readable name of the credential configuration object.
+	CredentialsName string `json:"credentials_name" tf:"force_new"`
+	// The Amazon Resource Name (ARN) of the cross account role.
+	RoleArn string `json:"role_arn" tf:"force_new"`
+	// Time in epoch milliseconds when the credential was created.
+	CreationTime int64 `json:"creation_time,omitempty" tf:"computed"`
+	// Databricks credential configuration ID.
+	CredentialsId string `json:"credentials_id,omitempty" tf:"computed"`
+	// The external ID that needs to be trusted by the cross-account role. This
+	// is always your Databricks account ID.
+	ExternalId string `json:"external_id,omitempty" tf:"computed"`
+}
+
 func ResourceMwsCredentials() *schema.Resource {
 	p := common.NewPairSeparatedID("account_id", "credentials_id", "/")
 	return common.Resource{
 		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			accountID := d.Get("account_id").(string)
+			acc, err := c.AccountClient()
+			if err != nil {
+				return err
+			}
 			roleArn := d.Get("role_arn").(string)
 			credentialsName := d.Get("credentials_name").(string)
-			credentials, err := NewCredentialsAPI(ctx, c).Create(accountID, credentialsName, roleArn)
+
+			credentials, err := acc.Credentials.Create(ctx, provisioning.CreateCredentialRequest{
+				CredentialsName: credentialsName,
+				AwsCredentials: provisioning.CreateCredentialAwsCredentials{
+					StsRole: &provisioning.CreateCredentialStsRole{
+						RoleArn: roleArn,
+					},
+				},
+			})
 			if err != nil {
 				return err
 			}
-			d.Set("credentials_id", credentials.CredentialsID)
+			d.Set("credentials_id", credentials.CredentialsId)
+			_, hasAccountId := d.GetOk("account_id")
+			if !hasAccountId {
+				d.Set("account_id", c.Config.AccountID)
+			}
```

Comment:
If AccountID is always returned, we can save some lines by setting it without condition

Reply:
done